SU2_CFD/src/fluid/CCoolProp.cpp

```diff
@@ -56,28 +56,51 @@ void CCoolProp::SetTDState_rhoe(su2double rho, su2double e) {
   dTdrho_e = fluid_entity->first_partial_deriv(CoolProp::iT, CoolProp::iDmass, CoolProp::iUmass);
   dTde_rho = fluid_entity->first_partial_deriv(CoolProp::iT, CoolProp::iUmass, CoolProp::iDmass);
   if (fluid_entity->phase() == 6) {
+    // assume it is pure gas
+    fluid_entity->specify_phase(CoolProp::iphase_gas);
+    if (abs(Pressure / Pressure_Critical - 1) < 0.01) {
+      // if P is very cloase to Pc, the CoolProp has issue
+      Pressure = Pressure_Critical * 0.99;
+    }
```

Comment:
Can the pressure be higher than the critical pressure? Is the objective to put an upper bound or just to avoid P = Pc?

Reply:
Yes, the pressure can be higher or lower than critical pressure. 
The point is to avoid P=Pc or P is very close to Pc.

Reply:
Would it make more sense to have a switch here?  if Pressure is close to but above Pc, would be better to set to Pressure = 1.01* Pc?
  Will there be problems crossing through the critical pressure?

Reply:
You are right, it is better to add a switch here.

I added this switch to set P = 0.99 Pc or 1.01 Pc, depending P is lower or larger than Pc.